File: javafx/src/main/java/de/debuglevel/liberatepdf2/javafx/Application.java
package de.debuglevel.liberatepdf2.javafx;

import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Application extends javafx.application.Application {
	private static final Logger logger = LoggerFactory.getLogger(Application.class);

	public static void main(final String[] args) throws Exception {
		launch(args);
	}

	@Override
	public void start(final Stage stage) throws Exception {
		logger.info("Starting LiberatePDF2 JavaFX client...");

		final String fxmlFile = "/fxml/DropWindow.fxml";
		logger.debug("Loading FXML for main view from: {}...", fxmlFile);
		final FXMLLoader loader = new FXMLLoader();
		final Parent rootNode = loader.load(this.getClass().getResourceAsStream(fxmlFile));

		logger.debug("Showing JFX scene...");
		final Scene scene = new Scene(rootNode, 500, 500);
		scene.getStylesheets().add("/styles/styles.css");

		stage.setTitle("LiberatePDF2 JavaFX client");
		stage.setScene(scene);
		stage.show();
	}
}

File: javafx/src/main/kotlin/de/debuglevel/liberatepdf2/javafx/Application.kt
package de.debuglevel.liberatepdf2.javafx

import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage
import org.slf4j.LoggerFactory

class Application : javafx.application.Application() {
    private val logger = LoggerFactory.getLogger(Application::class.java)

    override fun start(stage: Stage) {
        logger.info("Starting LiberatePDF2 JavaFX client...")
        val fxmlFile = "/fxml/DropWindow.fxml"

        logger.debug("Loading FXML for main view from: {}...", fxmlFile)
        val loader = FXMLLoader()
        val rootNode = loader.load<Parent>(this.javaClass.getResourceAsStream(fxmlFile))

        logger.debug("Showing JFX scene...")
        val scene = Scene(rootNode, 500.0, 500.0)
        scene.stylesheets.add("/styles/styles.css")
        stage.title = "LiberatePDF2 JavaFX client"
        stage.scene = scene
        stage.show()
    }
}
